cedar-policy-core/src/ast/policy.rs

```diff
@@ -1289,8 +1363,29 @@ impl PrincipalOrResourceConstraint {
                 EntityIterator::One(euid)
             }
             PrincipalOrResourceConstraint::Eq(EntityReference::Slot) => EntityIterator::None,
+            PrincipalOrResourceConstraint::IsIn(_, EntityReference::EUID(euid)) => {
+                EntityIterator::One(euid)
+            }
+            PrincipalOrResourceConstraint::IsIn(_, EntityReference::Slot) => EntityIterator::None,
+            PrincipalOrResourceConstraint::Is(_) => EntityIterator::None,
         }
     }
+
+    /// Get an iterator over all of the entity type names in this constraint.
+    /// The Unspecified entity type does not have a `Name`, so it is excluded
+    /// from this iter.
+    pub fn iter_entity_type_names(&self) -> impl Iterator<Item = &'_ Name> {
+        self.iter_euids()
+            .filter_map(|euid| match euid.entity_type() {
+                EntityType::Concrete(name) => Some(name),
+                EntityType::Unspecified => None,
+            })
+            .chain(match self {
+                PrincipalOrResourceConstraint::Is(entity_type)
+                | PrincipalOrResourceConstraint::IsIn(entity_type, _) => Some(entity_type),
+                _ => None,
+            })
```

Comment:
does this double-count some entity types? does `self.iter_euids()` not already include this entity types?  if not, should it?  also, do we have tests that `.iter_entity_type_names()` does the right thing and doesn't double-count?

Reply:
This is only pulling out then entity type in `is` which isn't part of an euid, so it's not present in `iter_euids`.